src/main/java/ru/octol1ttle/flightassistant/computers/safety/ChunkStatusComputer.java

```diff
@@ -0,0 +1,166 @@
+    private static final float ADVICE_THRESHOLD = 1250f; // for transient
+    private static final float CAUTION_THRESHOLD = 3000f;
+    private static final float WARN_THRESHOLD = 5000f;
+
+    public ChunkStatusComputer(ComputerHost host, MinecraftClient mc, AirDataComputer data, TimeComputer time) {
+        this.host = host;
+        this.mc = mc;
+        this.data = data;
+        this.time = time;
+    }
+
+    @Override
+    public void tick() {
+        if (!data.isFlying()) {
+            return;
+        }
+
+        isLoaded = data.isCurrentChunkLoaded;
+
+        if (isLoaded) {
+            if (!host.faulted.contains(time) && time.prevMillis != null) {
+                lastEncounteredMS = time.prevMillis;
+            }
+            isLoaded = false;
+            offsetMS = 0f;
+        }
+
+        final boolean isSinglePlayerPause = (mc.isInSingleplayer() && mc.isPaused());
+        if (isSinglePlayerPause && !isLoaded) {
+            offsetMS = ((time.prevMillis - lastEncounteredMS) - lastDiffMS);
+        }
+
+        if (time.prevMillis != null && lastEncounteredMS > 0f) {
+            lastDiffMS = (time.prevMillis - offsetMS) - lastEncounteredMS;
+        }
+
+        currentStatus = calculateLevelStatus();
+    }
+
+    @Override
+    public String getId() {
+        return "chunk_state";
+    }
+
+    @Override
+    public void reset() {
+        currentStatus = ChunkStatusLevel.UNKNOWN;
+        recoverVector = new Vector2d(0.5, 0.1);
+        isLoaded = true;
+        lastDiffMS = Float.MIN_VALUE;
+        lastEncounteredMS = Float.MIN_VALUE;
+        offsetMS = 0;
+    }
+
+    public boolean shouldCorrectTerrain() {
+        return FAConfig.computer().unloadedChunkProtection.recover() && isWarning();
+    }
+
+    public boolean isLoaded() {
+        return isLoaded;
+    }
+
+    public float getLastDiffMS() {
+        return lastDiffMS;
+    }
+
+    public boolean isAdvisory() {
+        return currentStatus == ChunkStatusLevel.ADVISORY;
+    }
+
+    public boolean isCaution() {
+        return currentStatus == ChunkStatusLevel.CAUTION;
+    }
+
+    public boolean isWarning() {
+        return currentStatus == ChunkStatusLevel.WARNING;
+    }
+
+    public ChunkStatusLevel getStatus() {
+        return currentStatus;
+    }
+
+    public ChunkStatusLevel calculateLevelStatus() {
+        if (!data.isCurrentChunkLoaded) {
+            if (lastDiffMS >= WARN_THRESHOLD) {
+                return ChunkStatusLevel.WARNING;
+            } else if (lastDiffMS >= CAUTION_THRESHOLD) {
+                return ChunkStatusLevel.CAUTION;
+            } else if (lastDiffMS >= ADVICE_THRESHOLD) {
+                return ChunkStatusLevel.ADVISORY;
+            }
+        }
+        return ChunkStatusLevel.NORMAL;
+    }
+
+
+    public Color getIndicator(ChunkStatusLevel maxLevel) {
+        return switch (currentStatus.getHighestLevel(maxLevel)) {
+            case ADVISORY -> FAConfig.indicator().advisoryColor;
```

Comment:
The advisory color is to show things that the user needs to follow. Examples are ECAM actions, flight directors and target altitude/heading. It is not to be used for warnings